Source/Diagnostics/SliceDiagnostic.cpp

```diff
@@ -0,0 +1,411 @@
+#include "SliceDiagnostic.H"
+#include <AMReX_MultiFabUtil.H>
+#include <AMReX_PlotFileUtil.H>
+#include <AMReX_FillPatchUtil_F.H>
+
+#include <WarpX.H>
+
+using namespace amrex;
+std::unique_ptr<MultiFab> CreateSlice( const amrex::MultiFab *mf, const amrex::Vector<Geometry> dom_geom, amrex::RealBox &slice_realbox, amrex::IntVect &slice_cr_ratio )
+{
+    std::unique_ptr<MultiFab> smf;
+    std::unique_ptr<MultiFab> cs_mf;
+
+    Vector<int> slice_ncells(AMREX_SPACEDIM);
+    int nghost = 1;
```

Comment:
nghost should be at-least 1 -- it could be greater -- but we need at-least one ghost cell. 
This is because, if the location of the slice lies does not align with the data points, then the value at the slice location is interpolated. The interpolation requires an additional data point which is stored in the ghost cell. Since only linear interpolation is implemented, I used a single ghost cell.